js/treetest.js

```diff
@@ -140,4 +235,53 @@ function setup(input_tasks,input_tree,input_selectableParents,input_closeSibling
 	tasks.set(0);
 	disableButton('#nextTaskButton');
 	bindNextButton();
+
+	socket.connect();
+	socket.emitPageLoadEvent();
+}
+
+function bindEvents() {
+	bindEmitOnActivateNode();
+	bindEmitOnOpenNode();
+	bindEmitOnCloseNode();
+	bindEmitOnTaskChanged();
+	bindEmitOnSubmitResponse();
+	bindEmitOnWindowVisibilityChanged();
+}
+
+function bindEmitOnActivateNode() {
+	// Fires when the user activates a node (by clicking on it)
+	$('#tree').on('activate_node.jstree', function (_, { node }) {
+		socket.emitActivateNodeEvent(node);
+	});
+}
+
+function bindEmitOnOpenNode() {
+	// Fires when the user opens a parent node to expose its children, either by
+	// activating the parent node itself, or by activating the button next to it
+	$('#tree').on('open_node.jstree', function (_, { node }) {
+		socket.emitOpenNodeEvent(node);
+	});
+}
+
+function bindEmitOnCloseNode() {
```

Comment:
I am wondering if we need that event. I am basically only interested by activating/expanding.